SnailWay/SnailWay.cs

```diff
@@ -8,18 +8,21 @@ public class SnailWay
     {
         public static int[] CalculateWay(int[][] grid)
         {
+            if(grid == null)
+            {
+                return null;
+            }
+
             List<int> snailWay = new List<int>();
-            var tempGrid = grid as IEnumerable<IEnumerable<int>>;
 
-            while (tempGrid.Count() != 0)
+            while (grid.Count() != 0)
             {
-                snailWay.AddRange(tempGrid.ElementAt(0));
-                tempGrid = tempGrid.Skip(1);
-                snailWay.AddRange((tempGrid.Select(line => line.Last())));
-                tempGrid = tempGrid.Select(line => line.SkipLast(1))
-                              .Reverse()
-                              .Select(line => line.Reverse());
+                snailWay.AddRange(grid[0]);
+                grid = grid.Skip(1).ToArray();
+                snailWay.AddRange(grid.Select(line => line[^1]));
+                grid = grid.Select(line => line.SkipLast(1).Reverse().ToArray()).Reverse().ToArray();
```

Comment:
`Reverse().ToArray()).Reverse().ToArray();` - what is that? At least one ToArray is redundant

Reply:
Actually not. Otherwise it's impossible to compile because Enumerable<Enumerable<int>> cannot be converted to int[][], backwards it will work.